codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -795,14 +899,34 @@ Recurses through registers and determines whether a contract has already been re
 the given name and address
 */
 let checkContainsRegisteredContractAddress = (self: t, ~contractName, ~contractAddress) => {
-  let allAddr = self->getAllAddressesForContract(~contractName)
+  let allAddr = self.baseRegister->getAllAddressesForContract(~contractName)
   allAddr->Set.String.has(contractAddress->Address.toString)
 }
 
 /**
 * Returns the latest block number fetched for the lowest fetcher queue (ie the earliest un-fetched dynamic contract)
 */
-let getLatestFullyFetchedBlock = (self: t) => self.latestFetchedBlock
+let getLatestFullyFetchedBlock = (self: t) => {
+  //Consider pending dynamic contracts when calculating the latest fully fetched block
+  //Since they are now registered lazily on query or update of the fetchstate, not when
+  //the register function is called
+  let minPendingDynamicContracts = self.pendingDynamicContracts->Belt.Array.reduce(None, (
+    acc,
+    contract,
+  ) => {
+    let {registeringEventBlockNumber} = contract
+    minOfOption(registeringEventBlockNumber - 1, acc)->Some
+  })
+
+  switch (self.baseRegister.latestFetchedBlock, minPendingDynamicContracts) {
+  | ({blockNumber}, Some(pendingDynamicContractBlockNumber))
+    if pendingDynamicContractBlockNumber < blockNumber => {
+      blockNumber: pendingDynamicContractBlockNumber,
+      blockTimestamp: 0,
+    }
+  | (baseRegisterLatest, _) => baseRegisterLatest
+  }
+}
```

Comment:
It used to be implicit that after a contract registeration the base register would contain the latest fetched block. But now there could be pending dynamic contracts so we need to check against those if there are any.